### PR TITLE
Proxy http headers

### PR DESCRIPTION
**Pull request by eeliu**
https://travis-ci.com/eeliu/pinpoint-c-agent/builds/85818049
 
support #22

### CODE REVIEW

File: pinpoint_php/php_common.cpp [hunk 1]
```diff
@@ -245,6 +245,36 @@ std::string get_rpc()
...
+bool get_proxy_http_header(std::string &value,int& type)
+{
+
+//    TYPE_APP = 1,
+//     TYPE_NGINX = 2,
+//     TYPE_APACHE = 3
+
+    std::string header ="";
+
+    if( (header = getNameFromHeaderMap(PP_NGINX_PROXY)) != "") {
```
Comment (ChenGXQQ) [inline pinpoint_php/php_common.cpp]: If nginx -> apache -> php ??

File: pinpoint_common/trace.cpp [hunk 2]
```diff
@@ -1323,6 +1324,84 @@ namespace Pinpoint
...
+                        boost::trim(app);
+                        tvalue.__set_stringValue(app);
+                    }
+                }
+                break;
+            case TYPE_NGINX:
+                {
+                    if(PARSE_STR(DSTR))
+                    {
+                        tvalue.__set_intValue2(atoi(pvalue));
```
[the remark this answers is not in the document]
Comment (eeliu) [reply pinpoint_common/trace.cpp]: #define PARSE_STR(name) ((pvalue = strstr(value.c_str(),(name))) != NULL && (pvalue += strlen(name)) != NULL)
cover it.

File: pinpoint_php/php_common.cpp (hunk 1 above)
Comment (eeliu) [reply pinpoint_php/php_common.cpp]: yes,  Pinpoint-ProxyNginx > Pinpoint-ProxyApache > Pinpoint-ProxyApp

File: pinpoint_common/trace.cpp [hunk 3]
```diff
@@ -1346,14 +1346,10 @@ namespace Pinpoint
 #define iSTR    "i="
 #define bSTR    "b="
 
-#define PARSE_STR(name) ((pvalue = strstr(value.c_str(),(name))) != NULL && (pvalue += strlen(name)) != NULL)
+#define PARSE_STR(name) ((pvalue = strstr(value.c_str(),(name))) != NULL && *(pvalue += strlen(name)) != '\0')
```
Comment (eeliu) [inline pinpoint_common/trace.cpp]: pvalue could be '\0'

**Pull request merged**